edit/edit.go

```diff
@@ -461,6 +476,146 @@ func ContainsComments(expr build.Expr, str string) bool {
 	return false
 }
 
+// RemoveEmptySelectsAndConcatLists iterates the tree in order to turn
+// empty selects into empty lists and adjacent lists are concatenated
+func RemoveEmptySelectsAndConcatLists(e build.Expr) build.Expr {
+	switch e := e.(type) {
+	case *build.BinaryExpr:
+		if e.Op == "+" {
+			e.X = RemoveEmptySelectsAndConcatLists(e.X)
+			e.Y = RemoveEmptySelectsAndConcatLists(e.Y)
+
+			x, xIsList := e.X.(*build.ListExpr)
+			y, yIsList := e.Y.(*build.ListExpr)
+
+			if xIsList && yIsList {
+				return &build.ListExpr{List: append(x.List, y.List...)}
+			}
+
+			if xIsList && len(x.List) == 0 {
+				return e.Y
+			}
+
+			if yIsList && len(y.List) == 0 {
+				return e.X
+			}
+		}
+	case *build.CallExpr:
+		if x, ok := e.X.(*build.Ident); ok && x.Name == "select" && e.List == nil {
+			return &build.ListExpr{List: []build.Expr{}}
+		}
+	}
+
+	return e
+}
+
+// SelectListsIntersection returns the intersection of the lists of strings inside
+// the dictionary argument of the select expression given as a parameter
+func SelectListsIntersection(sel *build.CallExpr, pkg string) (allEmpty bool, intersection []build.Expr) {
+	dict := sel.List[0].(*build.DictExpr)
+	firstList := dict.List[0].(*build.KeyValueExpr).Value.(*build.ListExpr).List
+	intersection = make([]build.Expr, len(firstList))
+	copy(intersection, firstList)
+	allEmpty = len(firstList) == 0
+
+	for _, keyVal := range dict.List[1:] {
+		keyVal := keyVal.(*build.KeyValueExpr)
```

Comment:
Although DictExpr nodes shouldn't have anything than KeyValueExpr as subnodes, in theory that's still possible, e.g. if there's a bug in another method in this file. It's safer to use a checked type cast and just `continue` if the current element is not what's expected, otherwise buildozer will just crash if something goes wrong.

Reply:
Wouldn't it be advisable to expose such a bug if it exists? Or print a warning if the type is different from KeyValueExpr.

Reply:
It makes sense, however it may indicate a bug somewhere else (especially if this module is used by a completely different project as a library, failing here would mean that the main project crashes.